Antidote/CallViewController.m

```diff
@@ -0,0 +1,184 @@
+//
+//  CallViewController.m
+//  Antidote
+//
+//  Created by Chuong Vu on 7/1/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "CallViewController.h"
+#import "Masonry.h"
+#import "Helper.h"
+#import "OCTCall.h"
+#import "OCTChat.h"
+#import "OCTSubmanagerCalls.h"
+
+NSString *const kCallViewControllerUserIdentifier = @"callViewController";
+
+@interface CallViewController () <RBQFetchedResultsControllerDelegate>
+
+@property (strong, nonatomic) OCTSubmanagerCalls *manager;
+@property (strong, nonatomic, readwrite) OCTChat *chat;
+@property (strong, nonatomic) OCTCall *call;
+@property (strong, nonatomic) RBQFetchedResultsController *callController;
+
+@property (strong, nonatomic) UILabel *nameLabel;
+@property (strong, nonatomic) UIButton *cancelCallButton;
+@property (strong, nonatomic) UILabel *timerLabel;
+
+@end
+
+@implementation CallViewController
+
+#pragma mark - Life cycle
+
+- (instancetype)initWithChat:(OCTChat *)chat submanagerCalls:(OCTSubmanagerCalls *)manager
+{
+    self = [super init];
+    if (! self) {
+        return nil;
+    }
+
+    _chat = chat;
+
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"chat == %@", chat];
+    _callController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall predicate:predicate delegate:self];
+    _callController.delegate = self;
+
+    _call = [_callController fetchedObjects].firstObject;
+    if (! _call) {
+        return nil;
+    }
+
+    _manager = manager;
+
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    self.view.opaque = YES;
+
+    [self setupBlurredView];
+    [self createNameLabel];
+
+    [self createEndCallButton];
+    [self createCallTimer];
+
+    [self layoutSubviews];
+}
+
+#pragma mark -  RBQFetchedResultsControllerDelegate
+
+- (void)controllerWillChangeContent:(RBQFetchedResultsController *)controller
+{}
+
+- (void)controller:(RBQFetchedResultsController *)controller didChangeObject:(RBQSafeRealmObject *)anObject atIndexPath:(NSIndexPath *)indexPath forChangeType:(NSFetchedResultsChangeType)type newIndexPath:(NSIndexPath *)newIndexPath
+{}
+
+- (void)controller:(RBQFetchedResultsController *)controller didChangeSection:(RBQFetchedResultsSectionInfo *)section atIndex:(NSUInteger)sectionIndex forChangeType:(NSFetchedResultsChangeType)type
+{}
+
+- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+{
+    self.call = [self.callController fetchedObjects].firstObject;
+
+    self.timerLabel.text = [self convertToTimeFromInterval:self.call.callDuration];
+    [self.timerLabel setNeedsDisplay];
+}
+#pragma mark - Private
+
+- (void)setupBlurredView
+{
+    self.view.backgroundColor = [UIColor clearColor];
+
+    UIView *darkView = [[UIView alloc] initWithFrame:self.view.bounds];
```

Comment:
What will happen on rotation of device? darkView and blurEffect would have wrong frame?

There is no need to set frame in init method, you can just set constraints for both of them to `make.edges.equalTo(self.view);`